repos/spack_repo/builtin/packages/sphexa/package.py

```diff
@@ -45,16 +58,21 @@ class Sphexa(CMakePackage, CudaPackage, ROCmPackage):
 
         depends_on("mpich +rocm", when="+rocm ^[virtuals=mpi] mpich")
 
-    conflicts("%gcc@:10")
+    conflicts("%gcc@:11", when="@0.95:")
+    conflicts("%gcc@:10", when="@:0.93.1")
     conflicts("cuda_arch=none", when="+cuda", msg="CUDA architecture is required")
     conflicts("amdgpu_target=none", when="+rocm", msg="HIP architecture is required")
     conflicts("+cuda", when="+rocm", msg="CUDA and HIP cannot both be enabled")
 
     def cmake_args(self):
         spec = self.spec
 
+        hdf5lib = "H5HUT"
+        with when("@:0.95"):
+            hdf5lib = "H5PART"
```

Comment:
please also double check the range: `@:0.95` means "up to 0.95 included" and 0.95 is a range version encompassing `0.95*`.

```suggestion
        hdf5lib = "H5HUT"
        if self.spec.satisfies("@:0.94"):
            hdf5lib = "H5PART"
```